src/Idris/IDEMode/CaseSplit.idr

```diff
@@ -60,37 +60,69 @@ toStrUpdate _ = pure [] -- can't replace non user names
 
 data UPD : Type where
 
+||| Given a list of definitions, a list of mappings from `RawName` to `String`,
+||| and a list of tokens to update, work out the updates to do, apply them, and
+||| return the result.
 doUpdates : {auto u : Ref UPD (List String)} ->
             Defs -> Updates -> List SourcePart ->
             Core (List SourcePart)
-doUpdates defs ups [] = pure []
+doUpdates defs ups [] = pure []   -- no more tokens to update, so we are done
+-- if we have an LBrace (i.e. `{`), handle its contents
 doUpdates defs ups (LBrace :: xs)
-    = let (ws, nws) = spanSpace xs in map (LBrace :: ws ++) $
-         case nws of
-           Name n :: RBrace :: rest =>
-                pure (Name n ::
-                      Whitespace " " :: Equal :: Whitespace " " ::
-                      !(doUpdates defs ups (Name n :: RBrace :: rest)))
-           Name n :: Equal :: rest =>
-                pure (Name n ::
-                      Whitespace " " :: Equal :: Whitespace " " ::
-                      !(doUpdates defs ups rest))
-           _ => doUpdates defs ups xs
+    -- the cases we care about are easy to detect w/o whitespace, so separate it
+    = let (ws, nws) = span isWhitespace xs in
+        case nws of
+          -- brace is immediately closed, so generate a new pattern-match on the
+          -- values the name can have, e.g. { x} where x : Nat would become
+          -- { x = Z}  (and later { x = (S k)})
+          Name n :: RBrace :: rest =>
+             pure (LBrace :: ws ++   -- preserve whitespace
+                   Name n ::
+                   Whitespace " " :: Equal :: Whitespace " " ::
+                   !(doUpdates defs ups (Name n :: RBrace :: rest))
+                   )
```

Comment:
Isn't this case handled by the similar one after line 87's `span isWhitespace rest`?

Reply:
Ah, so it is. Thanks for spotting that ^^;